satellite/satellitedb/migrate.go

```diff
@@ -1044,6 +1044,14 @@ func (db *DB) PostgresMigration() *migrate.Migration {
 						WHERE type=1 AND status=1 AND id=2;`,
 				},
 			},
+			{
+				Description: "Create partial index for user_credits table",
```

Comment:
This ought to have a comment saying why, since I think it's not obvious. Perhaps "This partial unique index enforces uniqueness among `(id, offer_id)` pairs for users that have signed up but are not yet activated (`credits_earned_in_cents`=0). Among users that _are_ activated, uniqueness of `(id, offer_id)` pairs is not required or desirable."